### PR TITLE
Themes: Remove Unused Current Theme Button

### PR DESCRIPTION
**Pull request by jeyip**
#### Changes proposed in this Pull Request

* Removes unused Current Theme Button file

#### Testing instructions

* Apply this PR
* Navigate to Appearance > Themes
* Verify that the current theme card at the top of the page and its associated actions still behave as expected

https://github.com/Automattic/wp-calypso/issues/60530

### CODE REVIEW

File: client/my-sites/themes/current-theme/button.tsx [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 import { Component } from 'react';
 
-export default class extends Component {
-	static displayName = 'CurrentThemeButton';
+interface CurrentThemeButtonProps {
+	name: string;
+	label: string;
+	icon: string;
+	href: string;
+	onClick: ( this: null, name: string ) => void;
```
Comment (zaguiini) [inline client/my-sites/themes/current-theme/button.tsx]: The first argument of `.bind` is the function context (i.e. `this` keyword).

It's not passed as an argument to the callback, so we can omit `this: null` from the declaration. It is not a parameter 😃

**Review by zaguiini — COMMENTED** [review]
Great stuff, count on me for TS tips and guidance 🚀

**Review by creativecoder — COMMENTED** [review]
Thanks for this! It's always nice to see PropTypes go away, in favor of TypeScript.

However, I don't think this component is used any longer. Looks like the only usage was removed in https://github.com/Automattic/wp-calypso/pull/55045. (But please double check, maybe I'm missing something.)

If that's the case, let's just delete it instead!

**Comment by jeyip** [comment]
> However, I don't think this component is used any longer. Looks like the only usage was removed in #55045. (But please double check, maybe I'm missing something.)

Yeah @zaguiini mentioned the same thing. I couldn't find references either, so I'm going to go ahead and remove the file 🙂

**Review by zaguiini — APPROVED** [review]
"Deleting code" means "less bugs". Thanks, bug fixer!

**Pull request merged**